packages/runtime/src/slate/BlockPlugin/index.tsx

```diff
@@ -100,10 +101,16 @@ export function onKeyDown(e: KeyboardEvent, editor: Editor) {
       return
     }
 
-    Transforms.splitNodes(editor, {
-      at: editor.selection,
-      always: true,
-      match: node => ElementUtils.isListItem(node),
+    Editor.withoutNormalizing(editor, () => {
+      if (!editor.selection) return
+
+      Transforms.splitNodes(editor, {
+        at: editor.selection,
+        always: true,
+        match: node => ElementUtils.isListItem(node),
+      })
+
+      Transforms.setNodes(editor, { key: uuid() }, { at: editor.selection })
```

Comment:
Basically, anytime we mutate the tree now we have to manually update the key field.  

In future commits I add a normalization that adds a key for you, but it doesn't replace existing keys so.

Reply:
By "mutate the tree" do you mean any change? This is something I had mentioned in a call we had last week and I thought that changing text of a node _didn't_ change its key

Reply:
"mutate the tree" here means splitting nodes, altering properties on them, etc. Changing the text will not update keys in this PR.

Reply:
Got it! Thanks for clarifying 👍🏻

Reply:
Altering properties on the nodes shouldn't change the key, though, right? Changing the text is equivalent to changing a property. For example, changing an inline or paragraph's color shouldn't generate a new key